Musicca/app/src/main/java/com/example/musicca/adapters/CurrentPlaylistAdapter.java

```diff
@@ -107,10 +105,9 @@ public int getItemCount() {
 
     // reuploads a list of items -- change to type used
     public void updateSongs() throws ParseException {
+        List<String> newSortedSongObjIds = sortSongObjectIds(sortedSongObjectIds);
         sortedSongObjectIds.clear();
         notifyDataSetChanged();
-
-        List<String> newSortedSongObjIds = sortSongObjectIds(songObjectIds);
         sortedSongObjectIds.addAll(newSortedSongObjIds);
```

Comment:
It seems you are adding the newly sorted list of songs back to the list. Doesn't this duplicate the songs in this list?

Reply:
It first clears all the songs in the playlist, through the function sortedSongObjectIds.clear(), notifies the adapter then list becomes empty. It seeks new songs from the database using the function sortSongObjectIds(String songObjectIds) in case of any new changes in the number of likes, it then adds all these songs back to the playlist again. The list does not duplicate

Reply:
Sorry I missed the 'clear()'. But can you make this more efficient by directly assign to newly sorted list to `sortedSongObjectIds`? That way you avoid copying the whole list.

Reply:
Since I removed the list songObjectIds, I am only working one list- sortedSongObjectIds. I am not sure how to approach this function without a place holder list, because once the sortedSongObjectIds list is cleared, it becomes empty and I cannot pass the empty list to the function sortSongObjectIds. 
For the adapter to realize that the list has changed, the list has to be cleared first, notified before adding new items and notified again.

Reply:
Okay I see what you mean. Thanks for the explanation!